server/fleet/policies.go

```diff
@@ -1,14 +1,54 @@
 package fleet
 
+// PolicyPayload holds data for policy creation.
+type PolicyPayload struct {
+	// QueryID allows creating a policy from an existing query.
+	//
+	// Using QueryID is the old way of creating policies.
+	// Use Query, Name and Description instead.
+	QueryID uint
+	// Name is the name of the policy (ignored if QueryID != 0).
+	Name string
+	// Query is the policy query (ignored if QueryID != 0).
+	Query string
+	// Description is the policy description text (ignored if QueryID != 0).
+	Description string
+	// Resolution indicate the steps needed to solve a failing policy.
+	Resolution string
+}
+
+// ModifyPolicyPayload holds data for policy modification.
+type ModifyPolicyPayload struct {
+	// Name is the name of the policy.
+	Name *string `json:"name"`
+	// Query is the policy query.
+	Query *string `json:"query"`
+	// Description is the policy description text.
+	Description *string `json:"description"`
+	// Resolution indicate the steps needed to solve a failing policy.
+	Resolution *string `json:"resolution"`
+}
+
+// Policy is a fleet's policy query.
+//
+// TODO(lucas): Add AuthorEmail mimicking Query.
 type Policy struct {
-	ID               uint    `json:"id"`
-	QueryID          uint    `json:"query_id" db:"query_id"`
-	QueryName        string  `json:"query_name" db:"query_name"`
+	ID uint `json:"id"`
+	// Name is the name of the policy query.
+	// NOTE(lucas): To not break clients (UI), I'm not changing this to json:"name" (#2595).
```

Comment:
We are breaking things anyway, let's make it `name`

Reply:
+1, I wanted to keep them for now as otherwise it completely breaks the "policies UI" and I wasn't able to test the changes.